File: problem-solving/programmers/level01/같은_숫자는_싫어.py
# https://programmers.co.kr/learn/courses/30/lessons/12906

def solution(arr):
    answer = []
    last = -1
    for i in arr:
        if last == i:
            continue
        answer.append(i)
        last = i
    return answer

# if __name__ == "__main__":
#     print(solution([1,1,3,3,0,1,1]))
#     print(solution([4,4,4,3,3]))

File: problem-solving/programmers/level01/같은_숫자는_싫어.java
// https://programmers.co.kr/learn/courses/30/lessons/12906

import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> tempList = new ArrayList<Integer>();
        int prevNum = 10;
        for (int num : arr) {
            if (num != prevNum) {
                tempList.add(num);
            }
            prevNum = num;
        }
        int[] answer = new int[tempList.size()];
        for (int i=0; i<answer.length; i++) {
            answer[i] = tempList.get(i).intValue();
        }
        return answer;
    }
}